src/service.js

```diff
@@ -454,20 +544,32 @@ function getServicePromiseInternal(holder, id) {
   if (cached) {
     return cached;
   }
-  const services = getServices(holder);
 
   // TODO(@cramforce): Add a check that if the element is eventually registered
   // that the service is actually provided and this promise resolves.
   let resolve;
   const p = new Promise(r => {
     resolve = r;
   });
-  services[id] = {
-    obj: null,
-    promise: p,
-    resolve,
-  };
-
+  const services = getServices(holder);
+  let s = services[id];
+  if (s) {
+    // Service is registered, but not yet instantiated.
+    s.promise = p;
+    s.resolve = resolve;
+    // Instantiate service immediately.
+    if (s.build) {
+      s.obj = s.build();
```

Comment:
It's admittedly a bit confusing. The call to getServicePromiseOrnullInternal will return a promise if the service has already been built, preventing this from executing more than once. I've added a test regardless.